networkx/algorithms/bipartite/covering.py

```diff
@@ -0,0 +1,67 @@
+"""
+********
+Covering
+********
+"""
+
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.utils import not_implemented_for, arbitrary_element
+
+
+@not_implemented_for('directed', 'multigraph')
+def min_edge_cover(G):
```

Comment:
This function is identical to the non-bipartite `min_edge_cover`, except for the particular matching algorithm used. Refactor them so that you have a single underlying edge covering algorithm that allows as input a maximum weight matching algorithm. Then this function simply calls that underlying algorithm and specifies the bipartite maximum weight matching algorithm, while the other simply calls the underlying algorithm and specifies the normal maximum weight matching algorithm.